RealEstate/apps/core/views.py

```diff
@@ -134,37 +133,15 @@ def post(self, request, *args, **kwargs):
         """
         homebuyer = Homebuyer.objects.filter(user_id=request.user.id)
         couple = Couple.objects.filter(homebuyer__user=request.user)
-        categories = Category.objects.filter(couple=couple)
         house = get_object_or_404(House.objects.filter(id=kwargs["house_id"]))
 
-        default_score = Grade._meta.get_field('score').default
-        for category in categories:
-            value = request.POST.get(str(category.id)) or default_score
-            grade, created = Grade.objects.update_or_create(
-                homebuyer=homebuyer.first(), category=category, house=house, defaults={'score': int(value)})
-
-        grades = Grade.objects.filter(house=house, homebuyer=homebuyer)
-
-        # Merging grades and categories to provide object with both information.
-        # Data Structure: [(cat1, score1), (cat2, score2), ...]
-        graded = []
-        for category in categories:
-            missing = True
-            for grade in grades:
-                if grade.category.id is category.id:
-                    graded.append((category, grade.score))
-                    missing = False
-                    break
-            if missing:
-                graded.append((category, None))
-        form = EvaluationForm(graded=graded)
-
-        messages.success(request,"Your evaluation was saved!")
-        context = {
-            'couple': couple,
-            'house' : house,
-            'grades': graded,
-            'form' : form
-        }
-        context.update(self._score_context())
-        return render(request, self.template_name, context)
+        if request.is_ajax():
+          id = request.POST['category']
+          score = request.POST['score']
```

Comment:
You might be right.  I haven't dug into the details.  From what I can tell the token stays the same between requests.  So it appears to be session related, meaning the user could just log in and grab the token value from the hidden input.  We are in agreement though, still a stretch goal.